ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -550,6 +551,22 @@ applyPatchNoMonoLocalBinds _ =
         (writeFile file . ("{-# LANGUAGE NoMonoLocalBinds #-}\n" ++))
         =<< readFile' file
 
+{- 'CmmParse.y' on the ghc-8.10.* branches is missing an import. It's
+ unclear why the stack ghc-lib build succeeds (and hence CI) but it
+ certainly does not build with cabal directly -- see
+ https://github.com/digital-asset/ghc-lib/issues/243 for where the
+ problem was first reported. On master, this file has moved to
+ GHC/cmm/Parser.y and already contains the missing directive so
+ nothing to do there. -}
+applyPatchCmmParseNoImplicitPrelude :: GhcFlavor -> IO ()
+applyPatchCmmParseNoImplicitPrelude ghcFlavor =
+  when (ghcFlavor == Ghc8101 || ghcFlavor == Ghc8102) $
+    writeFile "compiler/cmm/CmmParse.y" .
+      replace
+        "import GhcPrelude"
+        "import GhcPrelude\nimport qualified Prelude"
+    =<< readFile' "compiler/cmm/CmmParse.y"
```

Comment:
Is `readFile'` a strict version of `readFile` that closes the file before returning anything? Otherwise, I'd be surprised if this works.

Reply:
Yes. From https://hoogle.haskell.org/?hoogle=readFile%27: "A strict version of readFile. When the string is produced, the entire file will have been read into memory and the file handle will have been closed. Closing the file handle does not rely on the garbage collector."

Reply:
Thanks a lot for the explanation!